cve_bin_tool/helper_script.py

```diff
@@ -54,21 +49,32 @@ def __init__(self, filename, product_name=None):
         self.version_pattern = []
         self.vendor_product = self.find_vendor_product()
 
+        # for scanning files verisons
+        self.version_scanner = VersionScanner()
+
     def extract_and_parse_file(self, filename):
         """extracts and parses the file for common patterns, version strings and common filename patterns"""
 
         with self.extractor as ectx:
             if ectx.can_extract(filename):
                 binary_string_list = []
                 for filepath in self.walker([ectx.extract(filename)]):
-                    clean_path = VersionScanner.clean_file_path(filepath)
+                    clean_path = self.version_scanner.clean_file_path(filepath)
                     self.file_stack.append(f"{clean_path}")
                     # self.logger.warning(f"scanning binaries from {clean_path}")
 
                     # see if the file is ELF binary file and parse for strings
-                    if self.is_executable(filepath):
+                    try:
+                        # self.logger.debug(self.version_scanner.is_executable(filepath))
+                        is_exec, output = self.version_scanner.is_executable(filepath)
+                    except:
+                        is_exec = self.version_scanner.is_executable(filepath)
```

Comment:
I'm a bit confused about this `try except` block 🤔 
First of all `output` is never used so can be just `_`, but the main problem is in the usage of `is_executable`. At first I thought it was just a typo because you just call it in `try` AND `except` but expect a different number of return arguments, but then I looked at return type and (oh thanks type hinting!) and I see that it returns `True, output` or just `False`. But why do you need `try` here? Only then I understood that it's _because_ `is_executable` can return tuple or just value, urgh... Maybe it's some unknown to me pattern but I really don't like it, it's just confusing.

I think that something like this would be better:
```py
is_exec = self.version_scanner.is_executable(filepath)
if type(is_exec) == tuple:
    is_exec = is_exec[0]
```

But even better would be to at least change `is_executable` to return `False,` or `True, output`, then do `is_exec = self.version_scanner.is_executable(filepath)[0]`

Reply:
We've got the same try/except block in version_scanner.py where output *is* used in the case of python packages to do more processing, so whatever we do should be done in both places.  Good time for more refactoring!

Looking at this again, I'd go with changing `is_executable` so it returns `False, None` or `True, output` so it's always a tuple.  So in version_scanner.py you'd change the return False line in `is_executable` (line 118 of version.py) to be ...

```py
            return False, None
```

(Or maybe False, "" since output is otherwise a string, but None is probably more clear.)

Then later in version_scanner.py you'd use it as

```py
t, output = self.is_executable(filename)
``` 

And in helper_script.py, where you aren't ever using the output, you'd just ignore it:

```py
is_exec = self.version_scanner.is_executable(filepath)[0]
```

(note the `[0]` at the end)

with no try/except in either case.

Reply:
Ugh, and bonus refactor: let's change `t` in version_scanner.py to be `is_exec` -- no reason to use a one letter variable for this one.